src/openforms/forms/tests/variables/test_viewset.py

```diff
@@ -1320,3 +1326,272 @@ def test_api_call_returns_all_variables(self):
+        )
+        form_path = reverse("api:form-detail", kwargs={"uuid_or_slug": form.uuid})
+        form_url = f"http://testserver.com{form_path}"
+        form_definition_path = reverse(
+            "api:formdefinition-detail", kwargs={"uuid": form_definition.uuid}
+        )
+        form_definition_url = f"http://testserver.com{form_definition_path}"
+
+        data = [
+            {
+                "form": form_url,
+                "form_definition": form_definition_url,
+                "key": "profile",
+                "name": "Profile",
+                "data_type": FormVariableDataTypes.array,
+                "source": FormVariableSources.component,
+            },
+            {
+                "name": "profile-prefill",
+                "key": "profilePrefill",
+                "formDefinition": "",
+                "source": FormVariableSources.user_defined,
+                "prefillPlugin": COMMUNICATION_PREFERENCES_PLUGIN_IDENTIFIER,
+                "prefillAttribute": "",
+                "prefillIdentifierRole": "main",
+                "dataType": FormVariableDataTypes.string,
+                "prefillOptions": {
+                    "customerInteractionsApiGroup": customer_interactions_api.identifier,
+                    "profileFormVariable": "profile",
+                },
+                "form": form_url,
+            },
+        ]
+
+        response = self.client.put(
+            reverse(
+                "api:form-variables",
+                kwargs={"uuid_or_slug": form.uuid},
+            ),
+            data=data,
+        )
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+
+    def test_bulk_create_and_update_with_multiple_variables_using_different_prefill_options(
+        self,
+    ):
+        user = StaffUserFactory.create(user_permissions=["change_form"])
+        self.client.force_authenticate(user)
+
+        form = FormFactory.create()
+        form_step = FormStepFactory.create(
+            form=form,
+            form_definition__configuration={
+                "components": [
+                    {
+                        "type": "customerProfile",
+                        "key": "profileComponent",
+                        "name": "Profile component",
+                        "digitalAddressTypes": ["email"],
+                        "shouldUpdateCustomerData": True,
+                    },
+                    {
+                        "type": "customerProfile",
+                        "key": "secondProfileComponent",
+                        "name": "Second profile component",
+                        "digitalAddressTypes": ["phoneNumber"],
+                        "shouldUpdateCustomerData": True,
+                    },
+                ]
+            },
+        )
+        form_definition = form_step.form_definition
+        customer_interactions_api = CustomerInteractionsAPIGroupConfigFactory.create(
+            for_test_docker_compose=True
```

Comment:
this trait is used only when you need to request the Open Klant API (which would be accessible via docker compose), here it's not necessary.